File: frontend/src/backend/wails/configStorage.ts
import * as i from "../internal";
import {
  GetItem,
  RemoveItem,
  SetItem,
} from "./bindings/github.com/vovan-ve/sple-desktop/internal/backend/configstorage";

export const configStorage: i.ConfigStorage = {
  getItem: async (key: string): Promise<unknown> => {
    const r = await GetItem(key);
    return r ? JSON.parse(r.value) : undefined;
  },
  removeItem: RemoveItem,
  setItem: (key: string, value: unknown) => SetItem(key, JSON.stringify(value)),
};

File: internal/backend/config-storage.go
package backend

import (
	"github.com/vovan-ve/sple-desktop/internal/storage"
)

type ConfigStorage struct {
	F func() storage.Full[string]
}

func (s *ConfigStorage) GetItem(key string) (*StringRef, error) {
	v, ok, err := s.F().GetItem(key)
	if err != nil || !ok {
		return nil, err
	}
	return &StringRef{Value: v}, nil
}

func (s *ConfigStorage) SetItem(key string, value string) error {
	return s.F().SetItem(key, value)
}

func (s *ConfigStorage) RemoveItem(key string) error {
	return s.F().RemoveItem(key)
}

//func (s *ConfigStorage) GetAll() (map[string]string, error) {
//	return s.F().GetAll()
//}
//
//func (s *ConfigStorage) SetAll(all map[string]string) error {
//	return s.F().SetAll(all)
//}

type StringRef struct {
	Value string `json:"value"`
}
